File: Golang/remove-element.go
package main

func removeElement(nums []int, val int) int {
	var count int = 0

	for i := 0; i < len(nums); i++ {
		if nums[i] != val {
			nums[count] = nums[i]
			count++
		}
	}
	return count
}

func main() {
	nums := []int{3, 2, 2, 3}
	removeElement(nums, 3)
}

File: C++/remove-element.cpp
#include <vector>
#include <iostream>

class Solution {
public:
    int removeElement(std::vector<int>& nums, int val) {
        int count;
        for (size_t i = 0; i < nums.size(); i++)
        {
            if (nums[i] != val)
            {
                nums[count] = nums[i];
                count++;
            }
        }

            return count;
    }

};

int main()
{
    Solution x;
    std::vector<int> data = {3,2,2,3};

    x.removeElement(data,3);
    return 0;
}